pkg/trainer/replicas.go

```diff
@@ -171,15 +182,18 @@ func (s *TFReplicaSet) Create(config *tfv1alpha1.ControllerConfig) error {
 		}
 
 		log.Infof("Creating Service: %v", service.ObjectMeta.Name)
-		_, err := s.ClientSet.CoreV1().Services(s.Job.job.ObjectMeta.Namespace).Create(service)
+		createdService, err := s.ClientSet.CoreV1().Services(s.Job.job.ObjectMeta.Namespace).Create(service)
 
 		// If the job already exists do nothing.
 		if err != nil {
 			if k8s_errors.IsAlreadyExists(err) {
 				log.Infof("Service %v already exists.", s.jobName(index))
 			} else {
-				return k8sErrors.NewAggregate([]error{fmt.Errorf("Creating service %v returned error.", service.ObjectMeta.Name), err})
+				s.recorder.Eventf(s.Job.job, v1.EventTypeWarning, FailedCreateReason, "Error creating: %v", err)
+				return k8sErrors.NewAggregate([]error{fmt.Errorf("Creating service %v returned error.", createdService.ObjectMeta.Name), err})
 			}
+		} else {
```

Comment:
ditto.

Reply:
There is a subbranch https://github.com/tensorflow/k8s/pull/312/files/47249c0893c5be0c9e86f3f77ebbb0ea66c1726a#diff-d2bc8c1807fa25d2b911d2d781f48a07L178.

And if we remove else, there will be some idempotent events

Reply:
Yeah, I know. But if we remove the else, the branch that the service already exists will publish events, too.